ablate/common/models.py

```
@@ -13,11 +13,25 @@ def build_arm(arm, device="cuda", dtype=torch.float32, attn_impl="sdpa",
+              typed_attn_res_innovation_init=0.01):
     """arm in {'qwen','bibo_min'} -> (model, config). Params in `dtype` (fp32 master; bf16 via autocast).
     Balancing, each native: BiBo router-bias updates; Qwen Switch aux loss (aux_coef).
     PARAM MATCH: Qwen's num_experts is set to BiBo's GLU count, which is num_experts MINUS the
     param-free ±Identity specials -- see configs.glu_count."""
+    any_typed_feature = (
+        use_typed_attn_res
+        or use_typed_attn_res_fast_slow_memory
+        or use_typed_attn_res_innovation_write
+    )
+    if any_typed_feature and arm != "bibo_min":
+        raise ValueError("typed AttnRes is available only for the bibo_min arm")
```

Ruff (0.16.0)

[warning] Avoid specifying long messages outside the exception class

(TRY003)